kaggle_environments/envs/halite/halite.py

```diff
@@ -36,319 +38,133 @@ def get_to_pos(size, pos, direction):
-            del self.possible_ships[to_pos][ship_uid]
-
-
-def random_agent(obs, config):
-    size = config.size
-    halite = obs.halite
-    board = Board(obs, config)
-    player_halite, shipyards, ships = obs.players[obs.player]  
-
-    # Move, Convert, or have ships collect halite.
-    ships_items = list(ships.items())
-    # shuffle so ship1 doesn't always have first moving dibs.
-    shuffle(ships_items)
-    for uid, ship in ships_items:
-        pos, ship_halite = ship
-        # Collect Halite (50% probability when cell halite > ship_halite).
-        if board.shipyards[pos] == -1 and halite[pos] > ship_halite and randint(0,1) == 1:
-            continue
-        # Convert to Shipyard (50% probability when no shipyards, 5% otherwise).
-        if board.shipyards[pos] == -1 and player_halite >= config.convertCost and randint(0, 20 if len(shipyards) else 1) == 1:
-            player_halite -= config.convertCost
-            board.convert(uid)
-            continue
-        # Move Ship (random between all available directions).
-        move_choices = [None]
-        for direction in ["NORTH", "EAST", "SOUTH", "WEST"]:
-            to_pos = get_to_pos(size, pos, direction)
-            # Enemy shipyard present.
-            if board.shipyards[to_pos] != obs.player and board.shipyards[to_pos] != -1:
-                continue
-            # Larger ship most likely staying in place.
-            if board.ships[to_pos] is not None and board.ships[to_pos]["halite"] >= ship_halite:
-                continue
-            # Weigh the direction based on number of possible larger ships that could be present.
-            weight = 6
-            if board.ships[to_pos] is not None and board.ships[to_pos]["player_index"] == obs.player:
-                weight -= 1
-            for s in board.possible_ships[to_pos].values():
-                if s["halite"] > ship_halite:
-                    weight -= 1
-            move_choices += [direction] * weight
-        move = choice(move_choices)
-        if move is not None:
-            board.move(uid, move)
-            
-    # Spawn ships (30% probability when possible, or 100% if no ships).
-    for uid, pos in shipyards.items():
-        if board.ships[pos] is None and player_halite >= config.spawnCost and (randint(0, 2) == 2 or len(ships_items) == 0):
-            player_halite -= config.spawnCost
-            board.spawn(uid)
-
-    return board.action
+@board_agent
+def random_agent(board):
+    me = board.current_player
+    ships = me.ships
+    for ship in ships:
+        # None represents the chance to do nothing
+        ship.next_action = choice([ship_action for ship_action in ShipAction] + [None])
+    shipyards = me.shipyards
+    for shipyard in shipyards:
+        # 20% chance to spawn
+        shipyard.next_action = choice([ShipyardAction.SPAWN, None, None, None, None])
 
 
 agents = {"random": random_agent}
 
 
-def interpreter(state, env):
+def populate_board(state, env):
```

Comment:
Everything in the populate_board method is the same as it was before, just refactored into its own method. Reviewers can skip to the end of this method.